File: android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandlerRegistry.java
package com.swmansion.gesturehandler;

import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;

import java.util.ArrayList;

public interface GestureHandlerRegistry {
  ArrayList<GestureHandler> getHandlersForView(View view);
  View getViewForHandler(GestureHandler handler);
  @Nullable GestureHandler getHandler(int handlerTag);
}

File: packages/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/core/GestureHandlerRegistry.kt
package com.swmansion.gesturehandler.core

import android.view.View
import java.util.*

interface GestureHandlerRegistry {
  fun getHandlersForViewWithTag(viewTag: Int): ArrayList<GestureHandler>?
  fun getHandlersForView(view: View): ArrayList<GestureHandler>?
}
